src/goto-instrument/wmm/weak_memory.cpp

```diff
@@ -60,18 +60,23 @@ void introduce_temporaries(
        instruction.is_assert() ||
        instruction.is_assume())
     {
-      rw_set_loct rw_set(ns, value_sets, i_it
+      rw_set_loct rw_set(
+        ns,
+        value_sets,
+        function_id,
+        i_it
 #ifdef LOCAL_MAY
-      , local_may
+        ,
+        local_may
 #endif
       ); // NOLINT(whitespace/parens)
       if(rw_set.empty())
         continue;
 
       symbolt new_symbol;
```

Comment:
Unrelated to the purpose of this PR: I guess this should be using `util/fresh_symbol`.

Reply:
Certainly true, I'll take a look.

Reply:
Done in #3917.